docs/data/date-pickers/custom-field/BrowserV7MultiInputRangeField.tsx

```diff
@@ -8,18 +8,15 @@ import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
```

Comment:
- Use `DateRangePickerFieldProps` (with a small adaptation to work with multi input range field, I can create a `DateRangePickerMultiInputFieldProps` if you want that in the lib, my assumption that since we migrate away from multi input, it's not needed).
- Replace `BasePickersTextFieldProps` with `BaseMultiInputPickersTextFieldProps` (`BaseMultiInputPickersTextFieldProps` and `BaseSingleInputPickersTextFieldProps` are now incompatible so `BasePickersTextFieldProps` can't exist, and since we are migrating away from multi input I think we don't have to support this abstraction)
- Remove `defaultValue` and `onError` props which are never passed by the picker to the field so a field built to be used inside a picker doesn't have to support them